### ISSUE TITLE
I created a little batch script to update Brave Portable!

### ISSUE THREAD

**Issue by MagicalDrizzle** [portapps/brave-portable#153]
Script: https://github.com/MagicalDrizzle/batch-collection/blob/main/getbrave.cmd  
Instructions: https://github.com/MagicalDrizzle/batch-collection/blob/main/README.getbrave.md  
The script in action:
```
Select channel:
[1] Release
[2] Beta (default)
[3] Nightly
> 2

> Local: 1.63.126, Upstream: 1.63.126
> Up-to-date!

Press any key to continue . . .
```
```
Select channel:
[1] Release
[2] Beta (default)
[3] Nightly
> 1

> Local: 1.61.109, Upstream: 1.61.120
> New version available!
> Update? [Y,N]?
```
```
Select channel:
[1] Release
[2] Beta (default)
[3] Nightly
> 1

> Local: 1.63.126, Upstream: 1.61.120
> Warning: You may have chosen an update channel more downstream than your local version.
> For example, if you have Nightly version but you chose Beta/Release.
> Only proceed if you are switching your update channel.
> Do you wish to update anyway? [Y,N]?
```

**Comment by ednykissyou** [comment]
I won't use it, I'll save it as update.bat
After running, I selected 2.
If you press enter again, it will exit.
It seems like nothing has been updated.
I don't know if it's an update to "brake portable. exe" or an update to "brake portable \ app \ Brave. exe"

I hope to solve my confusion next time:
"Brave portable. exe" generated a data folder, but it started slowly.
Windows 11 still opens "brave portable \ app \ Brave. exe" by default, and Brave. exe is faster.
I really wish I could use "Brave. exe", but it won't generate a Data folder.
What should I do?

Thank you~

**Comment by MagicalDrizzle** [comment]
If you want to use `brave.exe`, 
```
start "" brave.exe --user-data-dir=profile --no-default-browser-check --disable-machine-id --disable-encryption-win
```
copy all of that into a file named `brave_portable.cmd` and put the file in the same directory as `brave.exe`.
I have created another script here, put it in the same location as `brave.exe` and run it to update Brave.
https://github.com/MagicalDrizzle/batch-collection/blob/main/getbrave_standalone.cmd

**Comment by mikuppy** [comment]
So we can use this script to update portable version without waiting for portapps ?

**Comment by MagicalDrizzle** [comment]
That's right :)

**Comment by mikuppy** [comment]
> That's right :) https://github.com/MagicalDrizzle/batch-collection/blob/main/getbrave_standalone.cmd If you use that script you won't even need to download brave-portable

Can you make a script for chrome as well

**Comment by MagicalDrizzle** [comment]
Sadly there does not seem to be any official offline installer download link for Chrome, you need to run the setup...the reason why this script is possible is Brave releasing the offline installer on GitHub.

If you are comfortable with Chromium instead, there is this project and I highly recommend it!
https://github.com/henrypp/chrlauncher

**Comment by Dutchgold** [comment]
yes the script installs brave to the portable folder but its no longer "portable" as the browsers data file now moves to the user folder hidden away in your pc,s local folder.

**Comment by JdotCarver** [comment]
- I followed the [provided instructions](https://github.com/MagicalDrizzle/batch-collection/blob/main/README.getbrave.md)
- While monitoring `%localappdata%` for any "Brave" entries.

 
### 🙌  Can confirm, the script works !
- Update to Brave.exe was done.
- Launching Brave-portable.exe works with the new binary.
- It appears as all extensions load fine.
- And contrary to what Dutchgold says : 
> yes the script installs brave to the portable folder but its no longer "portable" as the browsers data file now moves to the user folder hidden away in your pc,s local folder.

This was NOT the case as no local folder for Brave in `AppData/Local` was created.

 
### ➕ Addition
I added a feature to make it close brave if it is already opened, which I suggest you do.
```
@echo off
setlocal enabledelayedexpansion

:: Check if brave processes are running
tasklist /FI "IMAGENAME eq brave.exe" 2>NUL | find /I "brave.exe" > NUL
set BRAVE_RUNNING=%ERRORLEVEL%

tasklist /FI "IMAGENAME eq brave-portable.exe" 2>NUL | find /I "brave-portable.exe" > NUL
set BRAVE_PORTABLE_RUNNING=%ERRORLEVEL%

if %BRAVE_RUNNING% equ 0 (
    echo Brave is currently running.
    choice /C YN /M "Can I force close brave.exe?"
    if !errorlevel! equ 2 (
        echo Exiting without closing Brave.
        goto end
    ) else (
        taskkill /F /IM brave.exe > NUL
    )
)

if %BRAVE_PORTABLE_RUNNING% equ 0 (
    echo Brave Portable is currently running.
    choice /C YN /M "Can I force close brave-portable.exe?"
    if !errorlevel! equ 2 (
        echo Exiting without closing Brave Portable.
        goto end
    ) else (
        taskkill /F /IM brave-portable.exe > NUL
    )
)

:start
echo Select channel:
echo [1] Release (default)
echo [2] Beta
echo [3] Nightly

[the rest of the code...]
```

**Comment by Neustradamus** [comment]
A real solution must be added directly.
